zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/processinstance/ProcessInstanceResultRecord.java

```diff
@@ -99,6 +106,19 @@ public ProcessInstanceResultRecord setProcessInstanceKey(final long instanceKey)
     return this;
   }
 
+  @Override
+  public Set<String> getTags() {
+    final var tags = new HashSet<String>();
```

Comment:
🔧 You can use streams

```
    return tagsProperty.stream()
        .map(StringValue::getValue)
        .map(BufferUtil::bufferAsString)
        .collect(Collectors.toSet());
```

Reply:
I took this pattern from other records 🤔 thus I would leave it for now. But we could refactor all record in the future to use streams like you proposed.

Reply:
If you think we should use streams everywhere let's already change this, less work in the future 😄